apps/condo/domains/common/components/PostMessageProvider/globalHandlers.ts

```diff
@@ -138,4 +140,23 @@ export const useUpdateProgressBarHandler: () => RequestHandler<'CondoWebAppUpdat
         return { updated: true }
         // TODO(DOMA-5171): Adding miniAppTaskUIInterface in deps causing rerender hell!
     }, [userId])
+}
+
+export const useRedirectHandler: () => RequestHandler<'CondoWebAppRedirect'> = () => {
+    const router = useRouter()
+
+    return useCallback(({ url, target }) => {
+        let success = false
+        if (isSafeUrl(url)) {
+            if (target === '_blank' && typeof window !== 'undefined') {
+                window.open(url, target)
+                success = true
+            } else if (target === '_self') {
+                router.push(url)
```

Comment:
Can we check here, that origin is same as current window?
So only condo / v1 redirects will be allowed in same window....